lib/core/src/lib/testing/unit-testing-utils.ts

```diff
@@ -0,0 +1,477 @@
+        return this.loader.getHarness(MatInputHarness);
+    }
+
+    async getMatInputByCSS(selector: string): Promise<MatInputHarness> {
+        return this.loader.getHarness(MatInputHarness.with({ selector }));
+    }
+
+    async getMatInputByDataAutomationId(dataAutomationId: string): Promise<MatInputHarness> {
+        return this.loader.getHarness(MatInputHarness.with({ selector: `[data-automation-id="${dataAutomationId}"]` }));
+    }
+
+    async getMatInputByPlaceholder(placeholder: string): Promise<MatInputHarness> {
+        return this.loader.getHarness(MatInputHarness.with({ placeholder }));
+    }
+
+    async getMatInputHost(): Promise<TestElement> {
+        const input = await this.getMatInput();
+        return input.host();
+    }
+
+    async checkIfMatInputExists(): Promise<boolean> {
+        return this.loader.hasHarness(MatInputHarness);
+    }
+
+    async checkIfMatInputExistsWithCSS(selector: string): Promise<boolean> {
+        return this.loader.hasHarness(MatInputHarness.with({ selector }));
+    }
+
+    async checkIfMatInputExistsWithDataAutomationId(dataAutomationId: string): Promise<boolean> {
+        return this.loader.hasHarness(MatInputHarness.with({ selector: `[data-automation-id="${dataAutomationId}"]` }));
+    }
+
+    async checkIfMatInputExistsWithPlaceholder(placeholder: string): Promise<boolean> {
+        return this.loader.hasHarness(MatInputHarness.with({ placeholder }));
+    }
+
+    async clickMatInput(): Promise<void> {
+        const input = await this.getMatInput();
+        const host = await input.host();
+        await host.click();
+    }
+
+    async fillMatInput(value: string): Promise<void> {
+        const input = await this.getMatInput();
+        await input.setValue(value);
+    }
+
+    async fillMatInputByCSS(selector: string, value: string): Promise<void> {
+        const input = await this.getMatInputByCSS(selector);
+        await input.setValue(value);
+    }
+
+    async fillMatInputByDataAutomationId(dataAutomationId: string, value: string): Promise<void> {
+        const input = await this.getMatInputByDataAutomationId(dataAutomationId);
+        await input.setValue(value);
+        await (await input.host()).dispatchEvent('input');
+    }
+
+    async focusMatInput(): Promise<void> {
+        const input = await this.getMatInput();
+        await input.focus();
+    }
+
+    async blurMatInput(): Promise<void> {
+        const input = await this.getMatInput();
+        await input.blur();
+    }
+
+    async getMatInputValue(): Promise<string> {
+        const input = await this.getMatInput();
+        return input.getValue();
+    }
+
+    async getMatInputValueByDataAutomationId(dataAutomationId: string): Promise<string> {
+        const input = await this.getMatInputByDataAutomationId(dataAutomationId);
+        return input.getValue();
+    }
+
+    async sendKeysToMatInput(keys: string[] | TestKey[]): Promise<void> {
```

Comment:
Small thing but what if somebody would like to pass mixed sequence of keys like - characters which end with some special character? Maybe we should change type to the same type which is allowed by sendKeys: 
keys: (string | TestKey)[]
This type allows us to mix strings with TestKey inside single array

Reply:
Done